ra/registration-authority.go

```diff
@@ -116,7 +116,7 @@ func (ra *RegistrationAuthorityImpl) NewRegistration(init core.Registration) (re
 	// Store the authorization object, then return it
 	reg, err = ra.SA.NewRegistration(reg)
 	if err != nil {
-		err = core.InternalServerError(err.Error())
+		err = core.MalformedRequestError(err.Error())
```

Comment:
As far as I can tell by this point we have checked all the fields which constitute the `Authorization` object we construct so the only errors returned *should* be internal SQL transaction ones, the full body of which we may not want to pass back to the user?